src/gui/redux/docgen/docgen.js

```diff
@@ -0,0 +1,24 @@
+import {
+  REQUEST_DOCGEN,
+  RECEIVED_DOCGEN,
+  RECEIVED_EMPTY_DOCGEN,
+  FAILED_DOCGEN,
+} from './docgen.actions'
+
+export const EMPTY_DOCGEN = {}
+export const LOADING_DOCGEN = {}
```

Comment:
I just liked the way it simplifies reading & writing reducers. 
But I understand your point of view, I change that.